### PR TITLE
Introduce DOVECOT_INET_PROTOCOLS env

### LINKED ISSUE #2293: Letsencrypt subdomain error
# Subject

Hi I'm getting the error below everything seems to be working OK but, I was wondering if I should fix this.

I never seen this before and it just happened on a new install with a subdomain. Is this a subdomain issue?

Again everything does seems to work correctly.

Thanks

I have questions about TLS/SSL/STARTTLS/OpenSSL

## Description

	Error: sed -i -r s|^(smtpd_tls_chain_files =).*|\1 /etc/letsencrypt/live/sub.domain.co/privkey.pem /etc/letsencrypt/live/sub.domain.co/fullchain.pem| /etc/postfix/main.cf
	Error: sed -i -r -e s|^(ssl_key =).*|\1 </etc/letsencrypt/live/sub.domain.co/privkey.pem| -e s|^(ssl_cert =).*|\1 </etc/letsencrypt/live/sub.domain.co/fullchain.pem| /etc/dovecot/conf.d/10-ssl.conf

```

### LINKED ISSUE #2356: how to set listen=* to disable ipv6 for dovecot
# Subject

I have questions about configuration of covecot in container

## Description

The conatiner keeps produce the error while running

I found the default configuration of dovecot is listen = *, :: 

and should be set to listen=* to disable ipv6 but I found no way to replace it.

I tried set POSTFIX_INET_PROTOCOLS=ipv4 in environment variable but it's only for profix. 

am I missing somthing? Thanks very much.

``` BASH
mailserver    | Jan  4 10:49:17 mail dovecot: master: Error: socket() failed: Address family not supported by protocol
mailserver    | Jan  4 10:49:17 mail dovecot: master: Error: service(imap-login): listen(::, 143) failed: Address family not supported by protocol
mailserver    | Jan  4 10:49:17 mail dovecot: master: Error: socket() failed: Address family not supported by protocol
mailserver    | Jan  4 10:49:17 mail dovecot: master: Error: service(quota-status): listen(::, 65265) failed: Address family not supported by protocol
mailserver    | Jan  4 10:49:17 mail dovecot: master: Fatal: Failed to start listeners
```

### PR DESCRIPTION
**Pull request by casperklein**
# Description

`DOVECOT_INET_PROTOCOLS` let you configure, wether dovecot should listen on all interfaces or on IPv4/IPv6 only.

Fixes #2356

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/content/config/environment.md [hunk 1]
```diff
@@ -230,11 +230,19 @@ This option has been added in November 2019. Using other format than Maildir is
...
+- **all** => Listen on all interfaces.
+- ipv4 => Listen only on IPv4 interfaces. Most likely you want this behind Docker.
+- ipv6 => Listen only on IPv6 interfaces.
 
-Note: More details in <http://www.postfix.org/postconf.5.html#inet_protocols>
+Note: More details at http://www.postfix.org/postconf.5.html#inet_protocols
+
+##### DOVECOT_INET_PROTOCOLS
+
+- **not set** => Listen on all interfaces
```
Comment (georglauterbach) [inline docs/content/config/environment.md:241]: Could we have this behave like `POSTFIX_INET_PROTOCOLS` and have `all` as the default value here?

File: mailserver.env [hunk 2]
```diff
@@ -226,14 +226,22 @@ REPORT_SENDER=
...
+# Note: More details at http://www.postfix.org/postconf.5.html#inet_protocols
 POSTFIX_INET_PROTOCOLS=all
 
+# Choose TCP/IP protocols for dovecot to use
+# **not set** => Listen on all interfaces
+# ipv4 => Listen only on IPv4 interfaces. Most likely you want this behind Docker.
+# ipv6 => Listen only on IPv6 interfaces.
+#
+# Note: More information at https://dovecot.org/doc/dovecot-example.conf
+DOVECOT_INET_PROTOCOLS=
```
Comment (georglauterbach) [inline mailserver.env:243]: Likewise, here.

File: docs/content/config/environment.md [hunk 3]
```diff
@@ -168,7 +168,7 @@ Enabled by ENABLE_POSTFIX_VIRTUAL_TRANSPORT. Specify the final delivery of postf
 
 - **empty**: fail
 - `lmtp:unix:private/dovecot-lmtp` (use socket)
-- `lmtps:inet:<host>:<port>` (secure lmtp with starttls, take a look at <https://sys4.de/en/blog/2014/11/17/sicheres-lmtp-mit-starttls-in-dovecot/>)
```
Comment (georglauterbach) [inline docs/content/config/environment.md:171]: Out of curiosity, why remove the `<>`?

Comment (casperklein) [reply docs/content/config/environment.md:171]: I asked myself the opposite: why keeping them. They don't have a purpose + we don't use them in other places in our documentation where URLs are used.

Comment (casperklein) [reply docs/content/config/environment.md]: However: "Without angle brackets, the URL isn't converted into a link in many markdown parsers.". I've added them again.

File: docs/content/config/environment.md (hunk 1 above)
Comment (casperklein) [reply docs/content/config/environment.md]: Done.

File: mailserver.env (hunk 2 above)
Comment (casperklein) [reply mailserver.env]: Done.

File: docs/content/config/environment.md (hunk 3 above)
Comment (georglauterbach) [reply docs/content/config/environment.md]: That's exactly what I figured as well :D Nice.

File: target/scripts/startup/setup-stack.sh [hunk 4]
```diff
@@ -1100,12 +1100,33 @@ function _setup_postfix_vhost
...
+  if [[ ${DOVECOT_INET_PROTOCOLS} == "ipv4" ]]
+  then
+    PROTOCOL='*' # IPv4 only
+  elif [[ ${DOVECOT_INET_PROTOCOLS} == "ipv6" ]]
+  then
+    PROTOCOL='[::]' # IPv6 only
+  else
+    # Unknown value, panic.
+    dms_panic__invalid_value 'DOVECOT_INET_PROTOCOLS' "${DOVECOT_INET_PROTOCOLS}"
+  fi
```
Comment (georglauterbach) [inline target/scripts/startup/setup-stack.sh:1125]: With the new default value of `all`, this needs small adjustments, or am I overlooking something here?:)

Comment (casperklein) [reply target/scripts/startup/setup-stack.sh:1125]: You do, see [here](https://github.com/casperklein/archive_docker-mailserver/blob/03d887a0c7c9f03754467c7b0f28a9126ec0435a/target/scripts/start-mailserver.sh#L114).

**Review by polarathene — APPROVED** [review]

File: target/scripts/startup/setup-stack.sh (hunk 4 above)
Comment (georglauterbach) [reply target/scripts/startup/setup-stack.sh:1125]: I see :D

**Review by georglauterbach — APPROVED** [review]
LGTM

**Review by wernerfred — APPROVED** [review]

**Pull request merged**